app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkTopicsCompletedFragmentTest.kt

```diff
@@ -327,31 +296,38 @@ class MarkTopicsCompletedFragmentTest {
       createMarkTopicsCompletedTestActivityIntent(internalProfileId)
     ).use {
       testCoroutineDispatchers.runCurrent()
-      scrollToPosition(position = 0)
       performItemCheckOnTopicSummaryListItem(itemPosition = 0)
-      scrollToPosition(position = 1)
       performItemCheckOnTopicSummaryListItem(itemPosition = 1)
-      scrollToPosition(position = 2)
       performItemCheckOnTopicSummaryListItem(itemPosition = 2)
-      scrollToPosition(position = 3)
       performItemCheckOnTopicSummaryListItem(itemPosition = 3)
       onView(withId(R.id.mark_topics_completed_all_check_box)).check(matches(isChecked()))
```

Comment:
For this & the other test suites--should we also be including this line at the end of the new tests but to verify that the 'complete all' checkbox is unselected?

Reply:
Not sure i follow @BenHenning  which line exactly? I refactored scrollToPosition into performItemCheckOnTopicSummaryListItem.

Reply:
@BenHenning i think i understand now but adding a check to assert if checkbox is unchecked is always failing for some reason.

Reply:
What issue are you running into @KevinGitonga, and what specific check are you trying? Sometimes Espresso checks can be unexpectedly finicky.

Reply:
How odd. I looked into this and I couldn't see any reason why they would be different.

Just for sanity, can you verify that the ``not(isChecked())`` version fails when expected?

Reply:
@BenHenning very odd, i tried again and the first line is running as expected. I would say some checks have a tendency of behaving weirdly for some reason.

Reply:
Hmm. It may be prudent to run it a few times @KevinGitonga to be sure that it isn't flaky.

That being said, I think using ``isNotChecked`` is preferred for parity with ``isChecked``.

Reply:
@BenHenning what i mean is that matches(not(isChecked() has been used while covering all the other similar checks in this particular class. I do not understand what you mean by saying that. isnotChecked has more parity with isChecked. Also i have noticed the same behavior it's like espresso will misbehave and fail when expected to pass on some instances, not all i have also noticed this behavior on other tests.

Reply:
Also after running i few times it may pass but on the next try fail. What i have noticed is that not(isChecked) has little tendency to fail compared to isNotChecked.

Reply:
I see. Does the ``not(isChecked())`` version ever fail @KevinGitonga? If not, I'm fine with going ahead with that.

Also, it help to add a ``testCoroutineDispatchers.runCurrent()`` right before the verification line--that might help avoid the flake.

Reply:
Okay @BenHenning i have added this line but i still notice that it is failing to on some instances on my machine, but on CI i think it is passing as expected. Based on results, not sure if CI run all tests even newly added ones.

Reply:
Hmm @KevinGitonga let's sync on next steps for this issue. We should dig into the flake to make sure we don't check in a flaky test, and there are good practices to follow for how to more easily do this.

Reply:
Resolving this as we're now discussing this on the main comment thread.